lib/fog/aws/credential_fetcher.rb

```diff
@@ -44,7 +42,14 @@ def fetch_credentials(options)
                   :WebIdentityToken => File.read(options[:aws_web_identity_token_file] || ENV.fetch("AWS_WEB_IDENTITY_TOKEN_FILE")),
                   :Version => "2011-06-15",
                 }
-                connection = options[:connection] || Excon.new(STS_GLOBAL_ENDPOINT, :query => params)
+
+                if ENV["AWS_STS_REGIONAL_ENDPOINTS"] == "regional" && ENV["AWS_DEFAULT_REGION"]
```

Comment:
Could we simplify this by using a single ENV var, maybe `STS_ENDPOINT`and allow this to override the global value? I feel like that may be easier to understand and use than having a regional on/off value and a distinct region value. It's certainly possible I'm missing something here though. Are there precedents on key names I should be aware of?

Reply:
This is where I had the biggest back and forth personally.

I created this logic based on the environment variable method of configuring this as listed in the AWS SDK documentation here: https://docs.aws.amazon.com/sdkref/latest/guide/setting-global-sts_regional_endpoints.html.
As documented, there are two main ways to define whether or not regional endpoints should be used.

I suppose there are two different ways to go about this:

1) using an sts_regional_endpoints option in the config file - This could leverage the region configuration that *should* already be there. regional + region = regional STS URL. if sts_regional_endpoints is not defined or is not a valid value then default to the global endpoint.

2) The method that I originally took a stab at, which is the AWS supported method for determining this config with environment variables.

I guess a third option could be both, with the config file taking precedence. but the more that I think about it, maybe it makes more sense to support this configuration through the config file rather depending on environment variables? Let me know your thoughts @geemus

Reply:
Ah, yeah I suspected there might be a precedent like that you were referring back to.

I think we have some flexibility here to decide. I think we almost always try to support credential files and/or passed in settings first as it's arguably a more universal use case. Then, we usually add ENV as well if it's something that needs to be easily overridden at runtime. So I think we should definitely support config files and passed config, and ENV would be a nice bonus (from my understanding of things). Then again, maybe this would always only be a runtime override? How does your expected usage stack up against these possibilities?